### PR TITLE
fix: Fix web assembly build output

### PR DESCRIPTION
**Pull request by saulecabrera**
### WHY are these changes introduced?
Ensure that build artifacts are scoped under `build/`

### WHAT is this pull request doing?
Ensure that the generated web assembly binary is placed under `./build`

### Update checklist

- [ ] I've added a CHANGELOG entry for this PR (if the change is public-facing)

### CODE REVIEW

File: lib/project_types/script/layers/infrastructure/assemblyscript_project_creator.rb [hunk 1]
```diff
@@ -56,7 +56,7 @@ def write_package_json
               },
               "scripts": {
                 "test": "asp --summary --verbose",
-                "build": "shopify-scripts-toolchain-as build --src src/script.ts --binary #{script_name}.wasm -- --lib node_modules --optimize --use Date="
+                "build": "shopify-scripts-toolchain-as build --src src/script.ts --binary build/#{script_name}.wasm -- --lib node_modules --optimize --use Date="
```
Comment (jeffcharles) [inline lib/project_types/script/layers/infrastructure/assemblyscript_project_creator.rb:59]: To doublecheck, will this create the build directory if it doesn't already exist?

Comment (saulecabrera) [reply lib/project_types/script/layers/infrastructure/assemblyscript_project_creator.rb:59]: Yes.

**Review by jacobsteves — APPROVED** [review]

**Comment by saulecabrera** [comment]
> Will this build directory be gitignored as well?

@SpencerHaan Not as part of this PR. If we ever introduce a version control initialization process I think it will make sense to ignore it, right now we don't assume that the script will be tracked with version control. We could anyhow, let developers know through docs, that it's safe to ignore given that on `push` the script is re-built regardless.

**Comment by andrewhassan** [comment]
@SpencerHaan The outputted build directory will be in the user's project directory. When we bootstrap a project, we don't require that git is used and I don't believe we maintain a .gitignore; version control on their project is up to the user.

**Review by andrewhassan — APPROVED** [review]

**Review by jeffcharles — APPROVED** [review]

**Pull request merged**